rcljava/src/test/java/org/ros2/rcljava/subscription/SubscriptionTest.java

```diff
@@ -36,6 +36,14 @@ public void accept(final std_msgs.msg.String msg) {}
     assertEquals(node.getHandle(), subscription.getNodeReference().get().getHandle());
     assertNotEquals(0, subscription.getNodeReference().get().getHandle());
     assertNotEquals(0, subscription.getHandle());
+    assertEquals(1, node.getSubscriptions().size());
+
+    // We expect that calling dispose should result in a zero handle
+    // and the reference is dropped from the Node
+    subscription.dispose();
```

Comment:
These checks for the subscription handle are good.  I think we should also add something similar for publisher handles.

Reply:
I've added similar checks for publishers, services, and clients in 7972639